### PR TITLE
Make tall code blocks scrollable

### LINKED ISSUE #1123: Make tall code blocks scrollable
Very tall code blocks make the thread hard to follow and requires a lot of scrolling. Ideally, users should just post a Gist link, but they don't. Instead, we could limit the height and make it scrollable.
 
Example: https://github.com/sindresorhus/gifski-app/issues/12#issuecomment-369313344

### PR DESCRIPTION
**Pull request by arungalva**
closes #1123

### CODE REVIEW

**Comment by sindresorhus** [comment]
This looks good!

It's smart to cut off in the middle of a line, but what if it's cut off on an empty line? Then it's not clear that there's more content. I think we should make it more obvious then. How about a slight bottom inset shadow to indicate there's more content? @bfred-it Thoughts?

File: source/content.css [hunk 1]
```diff
@@ -971,3 +971,13 @@ body > .footer li a {
 	background: #9c95b3;
 	cursor: pointer;
 }
+
+blockquote {
+	max-height: 30.5em;
+	overflow-y: auto;
+}
+
+pre {
```
Comment (sindresorhus) [inline source/content.css]: These can be combined:

```css
blockquote,
pre {
	max-height: 30.5em;
	overflow-y: auto;
}
```

Comment (bfred-it) [reply source/content.css]: It should also be limited to comments so that other regular files and readmes aren’t affected.

**Comment by bfred-it** [comment]
@sindresorhus it’s hard to do that just with CSS because there’s no way to say “show only when there’s overflow.” It’d need some *css trick*

**Comment by sindresorhus** [comment]
https://stackoverflow.com/questions/44793453/how-do-i-add-a-top-and-bottom-shadow-while-scrolling-but-only-when-needed
https://stackoverflow.com/questions/38542163/pure-css-scroll-shadows
https://stackoverflow.com/questions/27451202/css-shadow-for-before-and-after-on-a-scrollable-container

**Comment by bfred-it** [comment]
This should probably be merged with the Gist max-height just to have them both in the same place and have both of them with the "scroll shadow" that Sindre suggested.

**Comment by arungalva** [comment]
Haven't had a chance to check this out in a while. Adding a shadow would make it very clear that there is more content there if they scroll. Will look into it this weekend 👨‍💻

**Comment by arungalva** [comment]
Hey Guys, finally got a chance to work on this. Here is how it looks when there is an overflow and when there isn't. Let me know what you guys think about the size of the shadow
screen shot 2018-03-27 at 2 34 07 am
screen shot 2018-03-27 at 2 52 17 am

I have also changed the selector to the following in order to limit this to only comments.
```
.comment-body > .highlight > blockquote,
.comment-body > .highlight > pre {
}
```

File: source/content.css [hunk 2]
```diff
@@ -971,3 +971,32 @@ body > .footer li a {
 	background: #9c95b3;
 	cursor: pointer;
 }
+
+.comment-body > .highlight > blockquote,
+.comment-body > .highlight > pre {
+	max-height: 30.5em;
+	overflow-y: auto;
+	max-height: 30.5em;
+	overflow-y: auto;
```
Comment (bfred-it) [inline source/content.css]: Duplicate props

File: source/content.css [hunk 3]
```diff
@@ -971,3 +971,32 @@ body > .footer li a {
...
+	http://lea.verou.me/2012/04/background-attachment-local/ */
+	background:
+		/* Shadow covers */
+		linear-gradient(rgba(255,255,255,0), #f6f8fa 70%) 0 100%,
+
+		/* Shadows */
+		radial-gradient(50% 0, farthest-side, rgba(0,0,0,.2), rgba(0,0,0,0)),
+		radial-gradient(50% 100%,farthest-side, rgba(0,0,0,.2), rgba(0,0,0,0)) 0 100%;
+	background:
+		/* Shadow covers */
```
Comment (bfred-it) [inline source/content.css]: We don't need lines 983-991, they're for legacy browsers.

File: source/content.css [hunk 4]
```diff
@@ -971,3 +971,32 @@ body > .footer li a {
...
+		linear-gradient(rgba(255,255,255,0), #f6f8fa 70%) 0 100%,
+
+		/* Shadows */
+		radial-gradient(farthest-side at 50% 0%, rgba(0,0,0,0), rgba(0,0,0,0)),
+		radial-gradient(farthest-side at 50% 100%, rgba(0,0,0,.2), rgba(0,0,0,0)) 0 100%;
+	background-repeat: no-repeat;
+	background-size: 100% 40px, 100% 40px, 100% 14px, 100% 14px;
+
+	/* Opera doesn't support this in the shorthand */
+	background-attachment: local, local, scroll, scroll;
```
Comment (bfred-it) [inline source/content.css]: No Opera

**Comment by bfred-it** [comment]
Tests:

> short
> block
> quote

```
short
code
```

> long
> long
> long
> long
> long
> long
> long
> long
> long
> long
> long
> long
> long
> long
> long
> long
> long
> long
> long
> long
> long
> long
> long
> long
> long
> long
> long
> long
> long
> quote

```
long
long
long
long
long
long
long
long
long
long
long
long
long
long
long
long
long
long
long
long
long
long
code
```

**Comment by bfred-it** [comment]
@sindresorhus the only problem with those solutions is that they are `background`-based, so any image/`pre` in the `blockquote` will cover the shadow. Good enough?

**Comment by sindresorhus** [comment]
@bfred-it Yeah, this doesn't look very good:

screen shot 2018-04-21 at 12 05 27

Any ideas how we can make it work when the scrolled content has a background?

**Comment by bfred-it** [comment]
By using a `:after` element but there goes the "css trick" to make it appear.

File: source/content.css [hunk 5]
```diff
@@ -1045,6 +1045,33 @@ body > .footer li a {
 	cursor: pointer;
 }
 
+.comment-body blockquote,
+.comment-body pre {
+	position: relative; /* OctoLink compat: attach the purple balls to the scroll */
```
Comment (shroudedcode) [inline source/content.css]: It's [*OctoLink**er***](https://octolinker.github.io/), not *OctoLink*.

**Comment by bfred-it** [comment]
Related: https://github.com/dear-github/dear-github/issues/166

**Comment by bfred-it** [comment]
Closing this PR since the suggested solution is completely different

**Comment by bfred-it** [comment]
We're overthinking it. GitHub's own code embeds don't show any shadows. Same for StackOverflow ([example](https://stackoverflow.com/a/49034579/288906))

We should drop the shadow and merge this

**Comment by sindresorhus** [comment]
Looks good, but needs to be mentioned in the readme. (Mostly so that the GitHub Papercuts team can more easily borrow the feature).

**Comment by lukehefson** [comment]
> Looks good, but needs to be mentioned in the readme. (Mostly so that the GitHub Papercuts team can more easily borrow the feature).

lol so true

**Comment by bfred-it** [comment]
> Looks good, but needs to be mentioned in the readme

@arungalva are you still available to make these final changes to merge your feature? :)

**Comment by kanitw** [comment]
Given this is done directly in `source/content.css`, how can I disable this?

**Comment by kanitw** [comment]
NVM, just found I can override the CSS.

**Comment by dwelle** [comment]
For people's reference → to disable this feature, add this to the extension's custom css:

```
.comment-body blockquote,
.comment-body pre {
  max-height: none;
}
```

image

**Comment by chrisblossom** [comment]
@dwelle Is there an updated way to disable this? The custom css override is not working for me on firefox. Thanks!

**Comment by bfred-it** [comment]
> @dwelle Is there an updated way to disable this? The custom css override is not working for me on firefox. Thanks!

It works if you add `!important` 

I updated the example above

**Pull request merged**